arches/app/views/workflow_history.py

```diff
@@ -0,0 +1,75 @@
+import json
+
+from django.db import transaction
+from django.db.models import F, JSONField, Value
+from django.db.models.expressions import CombinedExpression
+from django.db.utils import IntegrityError
+from django.utils.translation import gettext as _
+from django.views.generic import View
+
+from arches.app.utils.permission_backend import user_is_resource_reviewer
+from arches.app.utils.response import JSONErrorResponse, JSONResponse
+from arches.app.models import models
+
+
+class WorkflowHistoryView(View):
+
+    def get(self, request, workflowid):
+        if not user_is_resource_reviewer(request.user):
+            return JSONErrorResponse(_("Request Failed"), _("Permission Denied"), status=403)
+        try:
+            workflow_history = models.WorkflowHistory.objects.get(workflowid=workflowid, user=request.user)
+        except models.WorkflowHistory.DoesNotExist:
+            workflow_history = {}
```

Comment:
Or we could return a 404 and then check for that on the frontend when initializing.

Reply:
I think what you have here is good.